runtime/gc_include/ObjectAccessBarrierAPI.hpp

```diff
@@ -436,10 +436,13 @@ class MM_ObjectAccessBarrierAPI
 	VMINLINE void
 	copyObjectFields(J9VMThread *vmThread, J9Class *objectClass, j9object_t srcObject, UDATA srcOffset, j9object_t destObject, UDATA destOffset)
 	{
+		bool hasReferences = J9CLASS_HAS_REFERENCES(objectClass);
 #if defined(J9VM_GC_ALWAYS_CALL_OBJECT_ACCESS_BARRIER)
-		vmThread->javaVM->memoryManagerFunctions->j9gc_objaccess_copyObjectFields(vmThread, objectClass, srcObject, srcOffset, destObject, destOffset);
-#else /* defined(J9VM_GC_ALWAYS_CALL_OBJECT_ACCESS_BARRIER) */
-		if (j9gc_modron_readbar_none != _readBarrierType) {	
+		if (hasReferences) {
```

Comment:
This check is not needed here, and in fact is more future proof to not have it. Also we may for example have a (non-GC functional) barrier that profiles accesses for primitive types.